Sources/AblyAssetTrackingCore/Model/ErrorInformation.swift

```diff
@@ -122,3 +122,20 @@ public class ErrorInformation: NSObject, Error, CustomNSError {
         self.href = nil
     }
 }
+
+extension ErrorInformation: Hashable {
```

Comment:
Do the types modified in this PR need to be classes? They seem like good candidates for structs to me. And then we can use the default implementations of `Hashable`, `Equatable`.

Reply:
I changed it to the struct, but we can't use the default implementation of `Equatable` and `Hashable` because
>It only works for structs or enums (not classes) and then only when each stored property of a struct or associated value of an enum is itself Equatable or Hashable.

and the `public let cause: Error?` isn't `Equatable/Hashable`

changed here 57b89b9e66ff99a2b259aa8c7c8ebe7ce7f78a24

Reply:
Sorry if I wasn't clear, but I meant changing classes to structs for all the classes touched by this PR, not just `ErrorInformation`.

As for `Equatable`, `Hashable` etc – is it just `ErrorInformation` where you can't use the default implementation? Could we use it for the other types in this PR?

And I wonder whether it even makes sense for `ErrorInformation` to conform to `Hashable`, if `Error` doesn't. What is the use case?

Reply:
It's used in tests, e.g.:
https://github.com/ably/ably-asset-tracking-swift/blob/3d09ee69d9d31fd08c96b3d379cfebd94bd2f006/Tests/PublisherTests/DefaultPublisher/DefaultPublisherTests.swift#L185
https://github.com/ably/ably-asset-tracking-swift/blob/3d09ee69d9d31fd08c96b3d379cfebd94bd2f006/Tests/PublisherTests/DefaultPublisher/DefaultPublisherTests.swift#L485
>Sorry if I wasn't clear, but I meant changing classes to structs for all the classes touched by this PR, not just ErrorInformation.

I would like to do this in separate PR, WDYT?

Reply:
> I would like to do this in separate PR, WDYT?

The reason I suggested doing it here is that it avoids us needing to add all of the `Equatable` / `Hashable` implementations, only to then remove them again in the next PR.

Reply:
> It's used in tests, e.g.

I think it's probably more important to consider whether it makes sense for a user of the library. I can't think of an occasion in which a user would want to compare two `ErrorInformation` values. I think that if it's just used by the tests, better to just have a helper function in the test file to do the comparison.

Reply:
@lukasz-szyszkowski did you have any thoughts on this?

Reply:
> @lukasz-szyszkowski did you have any thoughts on this?

yea, I'll remove Hashable/Equatable from the ErrorInformation struct and I'll create some helper for testing purposes.

Reply:
@lawrence-forooghian here a476609341dbb72b495098bc2127b4ac3bcddb58